lib/src/strongly_connected_components.dart

```diff
@@ -45,21 +45,45 @@ List<List<T>> stronglyConnectedComponents<T extends Object>(
   var index = 0;
   final lastVisited = Queue<T>();
 
-  void strongConnect(T node) {
-    indexes[node] = index;
-    var lowLink = lowLinks[node] = index;
-    index++;
-    lastVisited.addLast(node);
-    onStack.add(node);
-    for (final next in edges(node)) {
+  final stack = [for (final node in nodes) _StackState(node)];
+  outer:
+  while (stack.isNotEmpty) {
+    final state = stack.removeLast();
+    final node = state.node;
+    var iterator = state.iterator;
+
+    late int lowLink;
```

Comment:
It looks like we could just leave this as nullable too - although having it late in a sense might be less bug prone as we would get an exception on line 86 if we don't ever assign a value. Ultimately either way is ok with me.

I do have a general preference for nullable versus late, with explicit `!` assertions. We could add one of those on line 86 as well, to retain the check that we did assign something, even though the compiler/analyzer wouldn't require it for a simple equality check.

Reply:
It looks like I can actually just remove `late` and make this non-nullable and the compiler can tell it's being assigned.